src/categorical_algebra/CSets.jl

```diff
@@ -260,4 +260,31 @@ cocone_objects(diagram) = ob(diagram)
 cocone_objects(span::Multispan) = feet(span)
 cocone_objects(para::ParallelMorphisms) = SVector(codom(para))
 
+# Functorial Data migration
+###########################
+
+""" 
+Migrates an instance of one ACSet 
+given dictionaries representing a functor between theories
+"""
+
+function migrate!(Y::ACSet{CD}, X::ACSet, FOb::Dict, FHom::Dict) where {CD}
```

Comment:
Instead of using a pair of dictionaries, we should introduce a struct

```julia
struct FreeFunctor{C,D,T<:Dict,U<:Dict}<: AbstractFunctor
FOb::Dict
FHom::Dict
end

dom(f::FreeFunctor{C,D, T,U}) = C
codom(f::FreeFunctor{C,D, T,U}) = D
```
with appropriate constructors. Then we can have `FreeFunctor` as an `@instance` of Category so that you could build them up with pushouts and such.

Reply:
I'd prefer to keep this PR simple and defer this to another day. #220 already did a lot along these lines and I'd rather try to deal with all that in a more coordinated way.